src/backend/src/controllers/users.controllers.ts

```diff
@@ -41,14 +41,14 @@ export const getUserSettings = async (req: any, res: any) => {
 
 export const updateUserSettings = async (req: any, res: any) => {
   const userId: number = parseInt(req.params.userId);
-  if (!req.body || !req.body.defaultTheme) {
+  if (!req.body || !req.body.defaultTheme || !req.body.slackId) {
```

Comment:
from the ticket "change the validation for updateusersettings to be using express validator instead of in the function (see line 44 of users.controllers.ts)"

Reply:
add it in the `user.routes.ts` as like `body('slackId').isString().not().isEmpty().optional({ nullable: true})` because that will allow it to either be undefined or null which is what we want. if you set something to undefined in prisma, it does nothing. if you set it to null, it turns it to null in the db. this way the frontend can send undefined if we want no change and null if we want to set it back to not having a slack id

Reply:
Should the error just be a json similar to how risks.controllers.ts does it (ex. ` return res.status(400).json({ errors: errors.array() });`)?

Reply:
yeah the whole `  const errors = validationResult(req);
  if (!errors.isEmpty()) {
    return res.status(400).json({ errors: errors.array() });
  }` thing is at the start so that checks the express validator package stuff and if there were any errors as part of that it throws them

Reply:
so just put that at the top of the endpoint. then you actually ahve to add the validation in the users.routes.ts file